components/SQLEditor.go

```diff
@@ -81,3 +87,99 @@ func (s *SQLEditor) SetBlur() {
 	s.SetBorderColor(tcell.ColorWhite)
 	s.SetTextStyle(tcell.StyleDefault.Foreground(tcell.ColorWhite))
 }
+
+
+/*
+	THIS FUNCTION OPENS EXTERNAL EDITOR.
+
+	Q: WHY OPEN ANOTHER TERMINAL?
+	A: OPENING EDITORS LIKE VIM/NEOVIM REALLY MESSED UP INITIAL TERMINAL'S OUTPUT.
+*/
+
+func openExternalEditor(s *SQLEditor) string {
+	// Current folder as path of temporary file
+	path := "./lazysql.sql"
+
+	editor := getEditor()
+	terminal := getTerminal()
+
+	// Create a temporary file with the current SQL query content
+	content := []byte(s.TextArea.GetText())
+
+	/*
+	0644 Permission
+	* User: read & write
+	* Group: read
+	* Other: read
+	*/
+
+	err := os.WriteFile(path, content, 0644)
+	if err != nil {
+		return s.TextArea.GetText()
+	}
+
+	// Remove the temporary file with the end of function
+	defer os.Remove(path)
```

Comment:
Does it work?

I assumed it would have been written


```suggestion
	defer func() { os.Remove(path)}
```

Reply:
I'm sure it works.